Compilation error after nix-develop

### Just checking...

- [x] This is a `cardano-wallet` bug.
- [x] I am using the latest `cardano-wallet` release.
- [x] I am using the *correct* `cardano-node` version for that release of `cardano-wallet`.

### Version

master

### Platform

linux

### Installation method

Built from source

### Network configuration

_No response_

### Context

After getting to `nix-develop` there is the following compilation error (it is warning of redundant import)

### Description

## Steps to Reproduce

<!-- Steps to reproduce the behavior. -->

1. call `nix-develop`
2. call `just clean-build`
...

### Expected behavior

<!-- A clear and concise description of what you expected to happen. -->
Compilation without errors

### Actual behavior

<!-- A clear and concise description of what you observe
instead. If applicable, add screenshots to help explain your
problem. -->

```
$ just clean-build
# shellcheck disable=SC1083
cabal build all --enable-benchmarks --enable-tests --minimize-conflict-set -O0 -v0 --ghc-options="-Werror "
src/FlakyTests/GHA.hs:24:7: error: [GHC-38856] [-Wunused-imports, Werror=unused-imports]
    The import of 'mapMaybe' from module 'Data.Maybe' is redundant
   |
24 |     , mapMaybe
   |       ^^^^^^^^

Error: [Cabal-7125]
Failed to build flaky-tests-0.2025.12.15 (which is required by test:unit from flaky-tests-0.2025.12.15 and exe:flaky-tests from flaky-tests-0.2025.12.15).

error: Recipe `build` failed on line 36 with exit code 1
error: Recipe `clean-build` failed on line 51 with exit code 1
```

--- a/cabal.project
+++ b/cabal.project
@@ -53,7 +53,7 @@ index-state: 2026-01-29T00:00:00Z
 
 index-state:
   , hackage.haskell.org 2026-01-29T00:00:00Z
-  , cardano-haskell-packages 2026-01-28T00:00:00Z
+  , cardano-haskell-packages 2026-02-05T12:58:40Z
 
 packages:
   lib/address-derivation-discovery

--- a/flake.nix
+++ b/flake.nix
@@ -122,7 +122,7 @@
       flake = false;
     };
     customConfig.url = "github:input-output-hk/empty-flake";
-    cardano-node-runtime.url = "github:IntersectMBO/cardano-node?ref=10.5.3";
+    cardano-node-runtime.url = "github:IntersectMBO/cardano-node?ref=10.5.4";
     mithril = {
       url = "github:input-output-hk/mithril?ref=2543.1-hotfix";
       inputs.nixpkgs.follows = "nixpkgs-unstable";

--- a/lib/flaky-tests/src/FlakyTests/GHA.hs
+++ b/lib/flaky-tests/src/FlakyTests/GHA.hs
@@ -21,7 +21,6 @@ import Data.ByteString.Lazy
     )
 import Data.Maybe
     ( catMaybes
-    , mapMaybe
     )
 import Data.Proxy
     ( Proxy (..)

--- a/lib/network-layer/src/Cardano/Wallet/Network/Implementation.hs
+++ b/lib/network-layer/src/Cardano/Wallet/Network/Implementation.hs
@@ -176,8 +176,7 @@ import Control.Monad.Class.MonadST
     ( MonadST
     )
 import Control.Monad.Class.MonadThrow
-    ( MonadMask
-    , MonadThrow
+    ( MonadEvaluate
     )
 import Control.Monad.Class.MonadTimer
     ( MonadTimer
@@ -825,7 +824,7 @@ type WalletNodeToClientProtocols m =
 mkWalletClient
     :: forall m block
      . ( block ~ CardanoBlock (StandardCrypto)
-       , MonadThrow m
+       , MonadEvaluate m
        , MonadST m
        , MonadTimer m
        , MonadAsync m
@@ -860,7 +859,7 @@ mkWalletClient tr pipeliningStrategy follower cfg nodeToClientVer =
 mkFetchBlockClient
     :: forall m block
      . ( block ~ CardanoBlock (StandardCrypto)
-       , MonadThrow m
+       , MonadEvaluate m
        , MonadST m
        , MonadTimer m
        )
@@ -891,7 +890,12 @@ mkFetchBlockClient cfg blockQ nodeToClientVer =
 -- purposes of querying delegations and rewards.
 mkDelegationRewardsClient
     :: forall m
-     . (MonadST m, MonadTimer m, MonadIO m, MonadAsync m, MonadMask m)
+     . ( MonadST m
+       , MonadTimer m
+       , MonadIO m
+       , MonadAsync m
+       , MonadEvaluate m
+       )
     => Tracer m Log
     -- ^ Base trace for underlying protocols
     -> CodecConfig (CardanoBlock StandardCrypto)
@@ -934,7 +938,7 @@ mkWalletToNodeProtocols
        , MonadST m
        , MonadTimer m
        , MonadAsync m
-       , MonadMask m
+       , MonadEvaluate m
        )
     => Tracer m Log
     -- ^ Base trace for underlying protocols

--- a/scripts/release/release-candidate.sh
+++ b/scripts/release/release-candidate.sh
@@ -79,10 +79,10 @@ sed -i "s|WALLET_VERSION=.*|WALLET_VERSION=$NEW_GIT_TAG|g" README.md
 git commit -am "Update cardano-wallet version in README.md"
 
 sed -i "s|$OLD_GIT_TAG|$NEW_GIT_TAG|g" scripts/ci/ruby-e2e.sh
-git commit -am "Update cardano-wallet version in ruby-e2e.sh"
+git diff --quiet || git commit -am "Update cardano-wallet version in ruby-e2e.sh"
 
 sed -i "s|RELEASE_WALLET_TAG=.*|RELEASE_WALLET_TAG=$NEW_CABAL_VERSION|g" run/common/docker/run.sh
-git commit -am "Update cardano-wallet version in run/common/docker/run.sh"
+git diff --quiet || git commit -am "Update cardano-wallet version in run/common/docker/run.sh"
 
 RELEASE_COMMIT=$(git rev-parse HEAD)
 

